evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -511,3 +543,29 @@ def test_simple_imputer_ignores_natural_language(
         assert_frame_equal(result, X_df)
     elif df_composition == "single_column":
         assert_frame_equal(result, X_df)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        ["int col"],
+        ["float col"],
+        ["categorical col", "bool col"],
+        ["bool col", "float col"],
+        ["categorical col", "float col"],
+    ],
+)
+def test_simple_imputer_errors_with_bool_and_categorical_columns(
```

Comment:
Added this to cover the SimpleImputer moving over to not covering mixed cat/bool cols.